### ISSUE TITLE
Two issues raised by LGTM.com - need more details

### ISSUE THREAD

**Issue by AlexTereshenkov** [wemake-services/wemake-python-styleguide#576]
LGTM.com has found two issues in the project code base:

1. https://lgtm.com/projects/g/wemake-services/wemake-python-styleguide/snapshot/6f321524a3cd5ca4da7ead854765d87dcfc79b19/files/wemake_python_styleguide/types.py?sort=name&dir=ASC&mode=heatmap#L43 I see that you have used `# noqa: F401`. LGTM.com does support `noqa` comments (the comment should suppress raising alerts for the line) - but it failed in this particular case, I'm investigating. I was just wondering what is the reason for importing `Final` and `final` without using them - is it to work around some bug?

2. https://lgtm.com/projects/g/wemake-services/wemake-python-styleguide/snapshot/6f321524a3cd5ca4da7ead854765d87dcfc79b19/files/wemake_python_styleguide/visitors/base.py#L246 I am pretty sure this is a false positive because the linter wouldn't work if you couldn't initialize `BaseTokenVisitor`. However I am not able to see that `BaseTokenVisitor` class implements `__call__ ` to be callable in the form

```
        return cls(
            options=checker.options,
            filename=checker.filename,
            file_tokens=checker.file_tokens,
        )
```

So I am curious how does it work that you can call the `cls()`? I'd love to figure this out to improve the query so that such a false positive won't be raised. Thank you!

**Comment by sobolevn** [comment]
1. We had problems with importing `final` long ago, while it was in `typing_extension`. I used dark magic to solve it. I would appreciate a PR with the fix. Probably just replace all imports to direct `typing_extension`
2. We use type there, not instance. See `cls: Type['BaseTokenVisitor']` annotation. And when we `return cls(` we just fire the `__init__` method there

Thanks for helping with the project! Awesome service: lgtm, never heard about it 🙂

**Comment by AlexTereshenkov** [comment]
This makes sense, thank you. I'll work on that false positive raised.

**Comment by AlexTereshenkov** [comment]
> Thanks for helping with the project! Awesome service: lgtm, never heard about it slightly_smiling_face

I am part of the team that runs LGTM at Semmle - there is a rule you've implemented that LGTM has [BaseExceptionViolation](https://github.com/wemake-services/wemake-python-styleguide/blob/153da5d2958e6b3a192aea61194d3e1e8fda0efa/wemake_python_styleguide/violations/best_practices.py#L556) :)

**Comment by sobolevn** [comment]
@AlexTereshenkov awesome! 🌟

**Comment by sobolevn** [comment]
Refactored. New release is coming.
